sympy/matrices/eigen.py

```diff
@@ -456,18 +480,23 @@ def _bidiagonal_decomposition(M, upper=True):
     2. Complex Matrix Bidiagonalization : https://github.com/vslobody/Householder-Bidiagonalization
 
     """
+    method = method.lower()
+    if method not in ("lhc", "householder"):
+        raise ValueError("Invalid Method for Bidiagonalization")
 
     if type(upper) is not bool:
         raise ValueError("upper must be a boolean")
 
     if not upper:
-        X = _bidiagonal_decmp_hholder(M.H)
-        return X[2].H, X[1].H, X[0].H
-
-    return _bidiagonal_decmp_hholder(M)
+        X = M.H.bidiagonal_decomposition()
+        return X[2].T.C, X[1].T.C, X[0].T.C
+    if method == "lhc":
```

Comment:
When `_bidiagonal_decmp_hholder` is computed after the QR decomposition, then the original `Q` from the QR decomposition can be multiplied to the first factor to give a new orthogonal matrix for the decomposition.
I'd expect that when bidiagonal decomposition is done after the QR decomposition, the first matrix should be identity matrix, so it will be easier.

Reply:
> When _bidiagonal_decmp_hholder is computed after the QR decomposition, then the original Q from the QR decomposition can be multiplied to the first factor to give a new orthogonal matrix for the decomposition.

This can be done but there is a problem, The dimensions of the bidiagonalized matrix B are not the same as the dimensions of the input Matrix. Now when we just have to return the bidiagonalized matrix, it is not a big problem, as we can pad zeroes accordingly and the answer remains correct. But when we have to return the bidiagonalization, padding zeroes, is not the right way because even if it keeps everything right, the final Matrix U, which is ``Q * _U`` loses its orthogonality.

> I'd expect that when bidiagonal decomposition is done after the QR decomposition, the first matrix should be the identity matrix, so it will be easier.

Sadly this doesn't hold for cases of 3x3 or bigger,  only U1 is identity and U2,U3,.....Un need not be an identity matrix.

Reply:
Solution: I guess there are probably two different versions of QR, though I am not totally sure.
Here is something I did in Sympy and the equivalent in Matlab
```python

In [5]: A = Matrix(3, 3, range(1, 10))

In [6]: Q, R = A.QRdecomposition()

In [7]: Q.evalf(5)
Out[7]:
⎡0.12309  0.90453 ⎤
⎢                 ⎥
⎢0.49237  0.30151 ⎥
⎢                 ⎥
⎣0.86164  -0.30151⎦

In [8]: R.evalf(5)
Out[8]:
⎡8.124  9.6011   11.078⎤
⎢                      ⎥
⎣  0    0.90453  1.8091⎦

In [9]: Q * Q.T
Out[9]:
⎡5/6   1/3  -1/6⎤
⎢               ⎥
⎢1/3   1/3  1/3 ⎥
⎢               ⎥
⎣-1/6  1/3  5/6 ⎦

In [10]: Q.T * Q
Out[10]:
⎡1  0⎤
⎢    ⎥
⎣0  1⎦

```

Matlab:
```python
>> A = [ 1 2 3;4 5 6;7 8 9]

A =

     1     2     3
     4     5     6
     7     8     9

>> [Q,R] = qr(A)

Q =

   -0.1231    0.9045    0.4082
   -0.4924    0.3015   -0.8165
   -0.8616   -0.3015    0.4082


R =

   -8.1240   -9.6011  -11.0782
         0    0.9045    1.8091
         0         0   -0.0000

>> Q * Q.'

ans =

    1.0000    0.0000    0.0000
    0.0000    1.0000    0.0000
    0.0000    0.0000    1.0000

>> Q.' * Q

ans =

    1.0000   -0.0000   -0.0000
   -0.0000    1.0000    0.0000
   -0.0000    0.0000    1.0000
```

For a Matrix M to be orthogonal, 
M.T * M = I
M * M.T = I
and that I believe is a bug somewhere in QR in sympy, where we are getting correct R, but somehow an incomplete Q.

Reply:
I think that you should augment the `Q` with any basis that is linearly independent to original matrix and run orthogonalization again. (rather than simply padding with zero)

The idea behind I had made Q to be semi-orthogonal was because of skipping this redundant computation. So I may not consider that this is a bad result.
But the docs may be improved to give some rank-deficient examples.

```python3
from sympy import *

A = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
Q, R = A.QRdecomposition()
Q_aug = Matrix.hstack(Q, Matrix([0, 0, 1])).QRdecomposition()[0]
R_aug = Matrix.vstack(R, Matrix([0, 0, 0]).T)

print(Q_aug * Q_aug.T)
print(Q_aug.T * Q_aug)
print(Q_aug * R_aug)
```

Reply:
That's really great! I will submit a commit, just one problem remains, how can I find a basis linearly independent to the original matrix? Is there any function/or anything for that?

Reply:
Usually, you can try padding every standard basis
`Matrix([1, 0, 0])`, `Matrix([0, 1, 0])`, `Matrix([0, 0, 1])` and orthogonalize it until succeeds.
And mathematically, it's guaranteed to succeed because there should be any one of three can be linearly independent to the original basis.

But in my opinion, I'd rather have other open questions like whether extending the bidiagonalization to return a semi-orthogonal matrix like is possible or not.